cmd.go

```diff
@@ -0,0 +1,79 @@
+package gps
+
+import (
+	"bytes"
+	"fmt"
+	"os/exec"
+	"time"
+)
+
+// monitoredCmd wraps a cmd and will keep monitoring the process until it
+// finishes or a certain amount of time has passed and the command showed
+// no signs of activity.
+type monitoredCmd struct {
+	cmd     *exec.Cmd
+	timeout time.Duration
+	buf     *activityBuffer
+}
+
+func newMonitoredCmd(cmd *exec.Cmd, timeout time.Duration) *monitoredCmd {
+	buf := newActivityBuffer()
+	cmd.Stderr = buf
```

Comment:
The thing with having to separate buffers is that the result of `combinedOutput` will be unordered. We can return stderr first and then stdout or the other way around. But the order is lost. Or should we only return stdout if the command succeeds and stderr otherwise?